### ISSUE TITLE
Postgres: ALTER TYPE (name) ADD VALUE throws PRS (Found unparsable section)

### ISSUE THREAD

**Issue by ieggleton** [sqlfluff/sqlfluff#3449]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The following code:
```sql
ALTER TYPE foo ADD VALUE 'baz';
```

Generates the following error:
```
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: "ALTER
                       | TYPE foo ADD VALUE 'baz'"
```

### Expected Behaviour

Valid PostgreSQL syntax so no error should be thrown: https://www.postgresql.org/docs/9.1/sql-altertype.html

### Observed Behaviour

Please see 'What Happened'

### How to reproduce

Example SQL:
```sql
ALTER TYPE foo ADD VALUE 'baz';
```

```sh
sqlfluff lint --dialect postgres /path/to/example.sql
```

### Dialect

Postgres

### Version

0.13.2

### Configuration

No additional (default)

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
Looks like @greg-finley added some but not all types in #3265 with this comment:
> Adds the most common things for ALTER TYPE. It's probably enough to close https://github.com/sqlfluff/sqlfluff/issues/3216. I can add the remaining things when I get a chance.

If you are looking to contribute a fix, than just need to alter `AlterTypeStatementSegment` in `src/sqlfluff/dialects/dialect_postgres.py` and add a test case to `test/fixtures/dialects/postgres/postgres_alter_type.sql` and generate the YAML file and submit the PR.

**Comment by greg-finley** [comment]
> I can add the remaining things when I get a chance.

Haha whoops ...

**Comment by ieggleton-beamery** [comment]
Just spotted this(!)

I won't duplicate effort, although I had a few more test cases, e.g.
```sql
ALTER TYPE foo ADD VALUE 'baz';
ALTER TYPE foo ADD VALUE 'qux' BEFORE 'baz';
ALTER TYPE foo ADD VALUE 'quux' AFTER 'baz';
```
